lib/src/git_backend.rs

```diff
@@ -845,12 +845,11 @@ mod tests {
             author: create_signature(),
             committer: create_signature(),
         };
-        let commit_id1 = store.write_commit(&commit1).unwrap();
-        let mut commit2 = commit1;
+        let (commit_id1, mut commit2) = store.write_commit(commit1).unwrap();
         commit2.predecessors.push(commit_id1.clone());
         // `write_commit` should prevent the ids from being the same by changing the
         // committer timestamp of the commit it actually writes.
-        assert_ne!(store.write_commit(&commit2).unwrap(), commit_id1);
+        assert_ne!(store.write_commit(commit2).unwrap().0, commit_id1);
```

Comment:
It might be nice to test that the commit is correct as well, especially in the test where the timestamps change (so, that will test the following commit).

Reply:
Good idea. See how this looks.

Reply:
Another thing I would check there is that the commit id returned actually matches the commit returned. That would test that the timestamp manipulations you do are correct.

If this is less trivial than it seemed to me at first glance, we can skip it or postpone it.

Reply:
Ah, that's a good idea. Done. I also had to adjust the test case a bit because it ended up using negative timestamps, which libgit2 doesn't seem to preserve.